rocks-lib/src/lua_rockspec/serde_util.rs

```diff
@@ -68,3 +70,66 @@ where
         })
         .try_collect()
 }
+
+pub(crate) enum DisplayLuaValue {
+    // NOTE(vhyrro): these are not used in the current implementation
+    // Nil,
+    // Number(f64),
+    Boolean(bool),
+    String(String),
+    List(Vec<Self>),
+    Table(Vec<DisplayLuaKV>),
+}
+
+pub(crate) struct DisplayLuaKV {
+    pub(crate) key: String,
+    pub(crate) value: DisplayLuaValue,
+}
+
+impl Display for DisplayLuaValue {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            //DisplayLuaValue::Nil => write!(f, "nil"),
+            //DisplayLuaValue::Number(n) => write!(f, "{}", n),
+            DisplayLuaValue::Boolean(b) => write!(f, "{}", b),
+            DisplayLuaValue::String(s) => write!(f, "\"{}\"", s),
+            DisplayLuaValue::List(l) => {
+                writeln!(f, "{{")?;
+
+                for item in l {
+                    writeln!(f, "{},", item)?;
+                }
```

Comment:
suggestion: (perhaps for a follow-up PR).

We could use a crate like [`pretty`](https://docs.rs/pretty/latest/pretty/) that should be able to handle things like indentation
(not sure if this crate does that - the Haskell `pretty` package does).